### ISSUE TITLE
Subscribe to OPML URLs

### ISSUE THREAD

**Issue by sebmos** [brentsimmons/NetNewsWire#626]
Dave Winer put out a ["proposal"](https://github.com/scripting/Scripting-News/issues/113) for feed readers to support subscribing to OPML files, rather than just importing them. This could allow for a social approach to sharing feeds, i.e. subscribing to "Brent's Seattle blogs" and getting updates as that list evolves.

AFAIK, none of that social ecosystem exists, but this is a bit of a chicken-and-egg problem - why bother managing an OPML file if it ends up imported to a feed reader, and further updates are made from within that. (Might as well use a feed reader, and export a static file every now and then.)

There would have to be some UI to manage all OPML subscriptions, with their contents added as feeds. As the OPML URLs change, feeds get added and removed (removed only if they're only added through a single OPML subscription, and never manually.)
There could also be some UI to indicate if OPML files aren't reachable anymore.

**Comment by brentsimmons** [comment]
It’s an old proposal, actually, and some feed readers have implemented it in the past. (FeedDemon, for instance, had this feature.) I’ve wanted to do it for NetNewsWire since the 2.0 days.

But here’s the problem: people will still require syncing of read/unread/starred status. So how do you provide that in this case?

There are multiple possible answers, yes. The trick is picking the best one.

**Comment by sebmos** [comment]
The way I see that feature isn't really that broad at all - I don't think this should be a sync feature at all. In my mind, the usefulness here isn't for an individual user subscribing to their own OPML files in different feed readers (thus requiring syncing of read/unread/starred, etc.), but of different users subscribing to the same OPML file (with the earlier "Brent's Seattle blogs" example, it'd be different people interested in Seattle that would all subscribe to your hypothetical OPML file), where all that context wouldn't need to be shared.

But it's possible I'm missing the point & most users would use or expect it to behave differently.

**Comment by brentsimmons** [comment]
Very few people would use the feature unless they could also sync their personal read/unread/starred status.

Think about it this way: you’ve subscribed to an OPML file on your Mac and on your iPhone. You read some article on your Mac. Then you go to read articles on your iPhone. Do you want the article marked-as-read there? I think you do. (Most people do, at least.)

**Comment by sebmos** [comment]
I agree that people would expect read status (etc.) synced between devices, but I think they'd expect that independent of whether or not their feeds/articles come from a OPML file subscription, a static OPML import or manually added feeds.

In my mind, syncing read/unread/starred is a separate (and much more important) feature. (Which is, I assume, why you put FeedBin syncing on the 1.0 roadmap.)

**Comment by brentsimmons** [comment]
The problem is is that it’s not actually separate. Feedbin will sync read/unread status for feeds that you’ve subscribed to in Feedbin. (Other systems work the same way.)

The issue becomes this: say you subscribe to an OPML file on your Mac. Your Mac tells Feedbin to add a folder and some feeds. Fine.

Now say you’re on your iPhone. There’s no way for your iPhone to know that this folder and feeds come from an OPML subscription. So you start making changes (having forgotten about the OPML file) — you move feeds around, delete some, maybe add a feed to that folder.

All of those changes sync to Feedbin.

Then you come back to your Mac. Your Mac is the only place that knows about the OPML subscription. So it re-reads that OPML file and wipes out some (or all) of the changes you made on your iPhone.

This is terrible user experience.

Ideally, Feedbin (and other syncing systems) would support this feature. That’s the best way to solve the problem.

**Comment by sebmos** [comment]
Yeah, that's a fair point.

Do you plan to add any features to NNW that aren't supported by all syncing engines? I don't have a good view of what their respective feature sets are/how much they differ. In theory, the "On My Mac" account could support this feature exclusively (until Feedbin & others add support.)

Perhaps differently: Assuming the syncing issue could somehow be resolved - is this something you'd be interested in adding in general, or is it too specific?

**Comment by brentsimmons** [comment]
I’ve wanted to add this feature for 15 years. :) I just haven’t worked out the best way to do it yet.

**Comment by jeroenleenarts** [comment]
I would love this one as well.

**Comment by voxpelli** [comment]
Here’s an example of an OPML file that you want to subscribe to without having any of the read/unread/starred status synced: https://github.com/voxpelli/webring-opml#current-webrings

I see OPML subscriptions as subscribing to someone’s blog roll and having the subscriptions in a folder in my feed reader stay in sync with that blog roll. That way I will automatically receive new blog feeds as the owner of the blog roll updates their list.
